pennylane_qiskit/converter.py

```diff
@@ -706,20 +713,20 @@ def mp_to_pauli(mp, register_size):
         mp(Union[ExpectationMP, VarianceMP]): MeasurementProcess to be converted to a SparsePauliOp
         register_size(int): total size of the qubit register being measured
     """
-
-    # ToDo: I believe this could be extended to cover expectation values of Hamiltonians
-
-    observables = {"PauliX": "X", "PauliY": "Y", "PauliZ": "Z", "Identity": "I"}
-
-    pauli_string = ["I"] * register_size
-    pauli_string[mp.wires[0]] = observables[mp.obs.name]
-
-    # Qiskit orders wires in the opposite direction compared to PL
-    pauli_string.reverse()
-
-    pauli_string = ("").join(pauli_string)
-
-    return SparsePauliOp(pauli_string)
+    pauli_strings = []
+    coeffs = []
+    op = mp.obs
+
+    for pauli_term, coeff in op.pauli_rep.items():
+        pauli_string = ["I"] * register_size
+        coeffs.append(coeff)
+        for _, val in enumerate(pauli_term.wires):
```

Comment:
Yeah a list comprehension is a good idea. Custom labels are not supported because we don't know the order of the wires.

e.g. 

op = (qml.X(1) + qml.Y(1)) @ (qml.Z(3) + qml.Z("ancilla"))
register_size = 4

The ancilla qubit could be index = 0 or index = 2 but we can't know for sure from the information passed to the mp_to_pauli function. We could fix this by adding a `wire_order` kwarg and then translating `wire_order` to some `wire_map`, but I think that would be better as a separate function.

In general, this conversion only works assuming that 1. the wire_order is ascending and 2. the wires are all ints.